### PR TITLE
[FIXED JENKINS-28658] - Create named container fingerprints

### PR DESCRIPTION
**Pull request by oleg-nenashev**
The plugin will use container name if it is available. Otherwise it will fall-back to the container ID.

@reviewbybees

### CODE REVIEW

**Comment by tfennelly** [comment]
:+1:

**Comment by jtnord** [comment]
:-1: code has a faint whiff of something to me.

**Comment by jtnord** [comment]
:-1: code has a faint whiff of something to me.

File: src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityReportListenerImpl.java [hunk 1]
```diff
@@ -72,10 +72,11 @@ private void processEvent(@Nonnull DockerTraceabilityReport report) throws IOExc
                
         // Update containerInfo if available
         final InspectContainerResponse containerInfo = report.getContainer();
-        @CheckForNull Fingerprint containerFP = null;
+        final @CheckForNull Fingerprint containerFP;
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityReportListenerImpl.java]: this looks bad as it is unused outside of the if statement below, uninitialized outside of the if statement, and only initialised once within the if statement.
I'm somewhat suprised that it compiles and passed findbugs.

Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/traceability/core/DockerTraceabilityReportListenerImpl.java]: Currently we do not use this fingerprint outside, so there's no findbugs issue. I'll just move the definition to if statement

**Comment by jtnord** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
The test failed die to the known reordering issue

**Pull request merged**